fauna/http/httpx_client.py

```diff
@@ -69,14 +73,29 @@ def request(
           json=data,
           headers=headers,
       )
+
+      if self._logger.isEnabledFor(logging.DEBUG):
+        headers_to_log = request.headers.copy()
+        headers_to_log.pop("Authorization")
```

Comment:
We don't exclude the Authorization header in the JVM driver, if a user wanted to log this header, could they do so?

Reply:
We exclude it if it's debug, and include it if it's trace, right? We don't get to differentiate that here and since we can't, I think it's too dangerous.

I think users can find a way to dump their own secret in debug cases if they want, and log that themselves using logging.debug.

Reply:
The JVM driver always logs the auth header, it includes the request body if the level is set to trace, or "FINEST" in Java parlance.

Reply:
We had a discussion in standup today to try and get consensus for all the drivers, we decided:

1. At DEBUG (or equivalent) level, redact the Authorization header (i.e. `Authorization: hidden`)
2. At TRACE (or equivalent) level, log the header value.

Reply:
Ack, at DEBUG level, this is already done. There is no TRACE level and python advises against custom levels. I think we can accept a slight difference between the drivers in this case and let users log their own secrets if they wish.

Reply:
I think there are still opportunities for this to bite us, for example, say a user has code like:

```
log.debug(appconfig.fauna_secret)
fauna.client(secret=appconfig.fauna_secret)
```

But because this is Python, `fauna_secret` could be implemented as something like:

```
@property
def fauna_secret -> str:
    .. some buggy code that returns different results ...
```

In this case we would not be able to inspect the auth header on the client side, without monkey patching.

Reply:
That's not really true. The issue we aimed to address from the event was inspection of the response headers client-side to help inform the next step. We're accomplishing what we intended with this.